### PR TITLE
Feat/implement-refresh-tokens

### PR DESCRIPTION
**Pull request by laipz8200**
# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [ ] Please open an issue before creating a PR or link to an existing issue
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Description

- Add `ACCESS_TOKEN_EXPIRE_MINUTES` to `.env.example` files and `docker-compose.yaml` for setting access token validity.
- Update `get_account_jwt_token` to use the configured expiration time.
- Rename `OAuthConfig` to `AuthConfig` to encompass general authentication settings.
- Correct typographical error in GitHub OAuth configuration description.

Fixes 

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [ ] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# Testing Instructions

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: api/services/account_service.py [hunk 1]
```diff
@@ -84,10 +113,13 @@ def load_user(user_id: str) -> None | Account:
...
 
     @staticmethod
-    def get_account_jwt_token(account, *, exp: timedelta = timedelta(days=30)):
+    def get_account_jwt_token(account: Account) -> str:
         payload = {
             "user_id": account.id,
-            "exp": datetime.now(timezone.utc).replace(tzinfo=None) + exp,
+            "exp": int(
+                datetime.now(timezone.utc).timestamp()
+                + timedelta(minutes=dify_config.ACCESS_TOKEN_EXPIRE_MINUTES).total_seconds()
```
Comment (bowenliang123) [inline api/services/account_service.py]: Considering the expiration time is used in seconds, it's suggested to change the time unit of this config from minutes to seconds for better consistency and flexibility.

File: api/services/account_service.py [hunk 2]
```diff
@@ -698,3 +749,8 @@ def _get_invitation_by_token(cls, token: str, workspace_id: str, email: str) ->
 
             invitation = json.loads(data)
             return invitation
+
+
+def _generate_refresh_token(length: int = 64):
```
Comment (bowenliang123) [inline api/services/account_service.py:753]:
```suggestion
def _generate_refresh_token(length: int = 64)->str:
```

File: api/services/account_service.py (hunk 1 above)
Comment (laipz8200) [reply api/services/account_service.py]: Thank you for your suggestion! For this scenario, using minutes is sufficient and more readable, as there won't be any access tokens with a validity period of less than one minute.

File: api/services/account_service.py (hunk 2 above)
Comment (laipz8200) [reply api/services/account_service.py:753]: Considering that in this situation the LSP has sufficient type inference capabilities, I personally lean towards not adding this annotation.

File: api/services/account_service.py (hunk 1 above)
Comment (bowenliang123) [reply api/services/account_service.py]: SGTM.

**Review by iamjoel — APPROVED** [review]

**Pull request merged**